### PR TITLE
Introduce a CoqIDE option to deactivate unicode binding completetion.

### LINKED ISSUE #14856: [CoqIDE] Unicode input issue when Maj+Space is bound to unbreakable space
#### Description of the problem
When using CoqIDE and entering the "\\" character followed by a (normal) space, the "Δ" character replaces both characters.
In #14501, @Zimmi48 reports the same issue, and links it to the presence of the Maj+Space key combination being bound to unbreakable space. I can confirm that I also have the same key combination bound, and that changing the keyboard configuration to one without that key combination solves the problem. It seems like the options of the IDE do not provide the possibility to opt out of the unicode input mode, which would probably be enough to at least be able to use CoqIDE without this issue.
Even if this issue seems not very widespread, I still believe that it is problematic since this key combination has been adopted in the recent normalization efforts for French keyboards disposition by the AFNOR (I was not able to confirm with certainty for the Azerty norm since the precise document is behind a paywall, but it is at least the case with the Bépo disposition).

#### Coq Version

Tested with Coq 8.13.0 and 8.11.2

### PR DESCRIPTION
**Pull request by ppedrot**
Provides a better workaround to #14856.

### CODE REVIEW

**Review by Alizter — APPROVED** [review]
Cool

I tested it and it works.

I'll merge tomorrow.

**Comment by Zimmi48** [comment]
I don't get how this is a bug fix. There is a bug, which is the behavior of the Unicode input mode in the presence of certain key bindings. This PR gives more flexibility to the user and a better workaround than what we have today, but it doesn't fix the bug, does it?

**Comment by ppedrot** [comment]
> it doesn't fix the bug, does it?

The bug does not seem to come from CoqIDE proper, since the callback is triggered by any keyboard input that looks like `<Shift>Space`. I don't know why BEPO insecable space does this, but in any case the problem would rather lie in GTK, upon which we have zero control. Apart from changing the key combination, I don't see how we could fix this thus.

Also, as already mentioned during the original discussion, I find the unicode input feature both utterly useless and an anti-pattern. Input should be delegated to whatever environment the user is running, and CoqIDE should not reinvent a broken wheel. There are a lot of designated IME to input math through LaTeX mnemonics in the wild, why on earth should we have to reimplement a homegrown version of it? Doing so we're bound to have nasty interferences like the one at stake.

File: doc/changelog/09-coqide/14863-coqide-opt-unicode-binding.rst [hunk 1]
```diff
@@ -0,0 +1,6 @@
+- **Fixed:**
+  It is now possible to deactivate the unicode completion
+  mechanism
+  (`#14863 <https://github.com/coq/coq/pull/14863>`_,
+  fixes `#14856 <https://github.com/coq/coq/issues/14856>`_,
```
Comment (Zimmi48) [inline doc/changelog/09-coqide/14863-coqide-opt-unicode-binding.rst:5]: As is, this isn't sufficient to consider the issue fixed, so this mention should be removed or amended. Also, the way to disable the Unicode completion mechanism should be documented, both in the changelog, and in the documentation of the feature.

**Review by Zimmi48 — CHANGES_REQUESTED** [review]

File: doc/changelog/09-coqide/14863-coqide-opt-unicode-binding.rst (hunk 1 above)
Comment (ppedrot) [reply doc/changelog/09-coqide/14863-coqide-opt-unicode-binding.rst]: Done.

**Review by Zimmi48 — APPROVED** [review]

**Comment by Alizter** [comment]
Shall I merge tomorrow?

**Comment by Zimmi48** [comment]
The current state is fine to me. @charguer may want to have a look though.

**Comment by charguer** [comment]
I have fine having unicode bindings "off" by default, provided there is an easily accessible menu for turning it on. 
This is probably better than having users unable to type in a backslash without a mess at the next character.

> why on earth should we have to reimplement a homegrown version of it.
A few years back, I spent several day trying quite a fair number of unicode input mechanisms that were advertized as "usable" on Ubuntu, and none was delivering the experience I was looking for. If you know of tools that deliver the same typing efficiency and customizability as the one I implemented, and that is easy to install on ubuntu AND macOS, let me know.

(Btw, I look forward to see what's the most practical input mode for Coq+VSCode on Ubuntu. Hopefully, I won't be required to port my code to a VSCode plugin.)

**Comment by herbelin** [comment]
As far as my experience is concerned, there used to be an input method for CoqIDE implemented by Vincent G. long time ago It was too complicated to install and I renounced. The approach working out of the box provided by the current mechanism (without falling in #14856) is with this respect pretty convenient.

**Comment by Zimmi48** [comment]
> I have fine having unicode bindings "off" by default, provided there is an easily accessible menu for turning it on.

I don't think that this PR turns the mode off by default, does it?

**Comment by Alizter** [comment]
During my testing, it was on by default.

**Comment by ppedrot** [comment]
No, this PR does not turn the mode off by default. The default value is `true` which means "on".

**Comment by Alizter** [comment]
I will merge tomorrow if there are no objections,.

**Comment by Alizter** [comment]
@coqbot merge now

**Pull request merged**